### DISCUSSION TITLE
In the context of ESP8266, what do the letters 'ets' stand for?

### DISCUSSION THREAD

**Discussion by DarioGHub** [esp8266/Arduino#8578] [Q&A]
The letters 'ets' are used throughout the firmware, SDK and esp8266/Arduino code.
1) first letters in the boot message:
```
 ets Jan  8 2013,rst cause:2, boot mode:(3,7)
```
2) numerous define and function identifiers. Just [search](https://github.com/esp8266/Arduino/search?q=ets) the esp8266/Arduino distribution.
```
#define os_delay_us ets_delay_us
ets_printf_P(const char *str, ...)
```
3) filenames:
```
#include "ets_sys.h"
```
Tried Google, but nothing, excepting online dictionaries with hundreds of possibilities.
Also scoured ~10 Espressif PDFs.

Is it an acronym for Espressif Tensilica System?
Pedantic perhaps, but I'd like to know. A reference would be over the top.

**Comment by Rojuinex** [comment]
Based on [this comment](https://github.com/espressif/esp-idf/blob/4523f2d67465373f0e732a3264273a8e84a1a6d1/components/esp_rom/include/esp32/rom/ets_sys.h#L41) I think its safe to assume it stands for "Espressif Task Scheduler".

**Comment by DarioGHub** [comment]
Agreed. Thanks soon much for digging that reference out of the source.
